test/rebar3_hex_publish_SUITE.erl

```diff
@@ -0,0 +1,127 @@
+-module(rebar3_hex_publish_SUITE).
+
+-compile(export_all).
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+all() ->
+    [  publish_invalid_semver
+     , publish_no_descrip
+     , publish_empty_descrip
+     , publish_has_contributors
+     , publish_has_maintainers
+     , publish_no_licenses
+     , publish_empty_licenses
+     , publish_multi_errors
+    ].
+
+publish_invalid_semver(Config) ->
+    App = mock_app("bad_ver", Config),
+    Repo = maps:merge(hex_core:default_config(),
+			#{name => <<"foo">>, repo => <<"foo">>,
+			  api_url => <<"http://127.0.0.1:3000">>,
+			  repo_url => <<"http://127.0.0.1:3000">>,
+			  repo_verify => false}),
+    State = State = rebar_state:new(),
+    Exp = {error,
+            {rebar3_hex_publish,
+             [{invalid_semver,<<"bad_ver">>,"0.a.1b..0.-foo"}]}},
+    Exp = rebar3_hex_publish:publish(App, Repo, State),
```

Comment:
You can get assert macros from `-include_lib("stdlib/include/assert.hrl").` and use `?assertMatch(A,B)` instead and then rebar3 is able to format any badmatch in a more readable error during test runs.

Reply:
aight, fixed that up and added a few more test suites (start of them anyway) while we're here.

Reply:
Right now we're pulling in the assertions via eunit.hrl, I did that as that's how you had it in the original suite. Keep like this or change to just pull in the assert.hrl?